web/src/copy_and_paste.ts

```diff
@@ -535,10 +540,17 @@ export function paste_handler_converter(
                 span.classList.add("zulip-paste-parser-text-node");
                 span.textContent = child.textContent;
                 child.replaceWith(span);
+            } else if (child instanceof HTMLElement) {
+                // Recursively parse the children to scan for text nodes with katex siblings that can be spanified
+                spanify_katex_sibling_text_nodes(child);
             }
         }
     }
 
+    for (const element of para_elements) {
+        spanify_katex_sibling_text_nodes(element);
+    }
```

Comment:
This seems to be adding a ton of complexity and I don't really think it's plausibly on a path to being 100% correct. Usually, a good parser should be simple, like most of the existing ones here. 

I think we need to look at rethinking the algorithm entirely. I don't think this spanification process should be required at all, if we use the right outer selectors for our walk. Do you a simple proof that we need to include it?